service/view_building_coordinator.cc

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <exception>
+#include <iterator>
+#include <ranges>
+#include <seastar/core/coroutine.hh>
+
+#include "cql3/query_processor.hh"
+#include "cql3/untyped_result_set.hh"
+#include "db/schema_tables.hh"
+#include "db/system_keyspace.hh"
+#include "schema/schema_fwd.hh"
+#include "seastar/core/loop.hh"
+#include "seastar/coroutine/maybe_yield.hh"
+#include "service/query_state.hh"
+#include "service/raft/group0_state_machine.hh"
+#include "service/raft/raft_group0.hh"
+#include "service/raft/raft_group0_client.hh"
+#include "utils/assert.hh"
+#include "utils/log.hh"
+#include "seastar/core/abort_source.hh"
+#include "seastar/core/with_scheduling_group.hh"
+#include "service/migration_manager.hh"
+#include "replica/database.hh"
+#include "view_info.hh"
+
+#include "service/view_building_coordinator.hh"
+
+logging::logger vbc_logger("vb_coordinator");
+
+namespace service {
+
+using view_name = db::system_keyspace_view_name;
+using vbc_tasks = db::system_keyspace_vbc_tasks;
+using vbc_view_tasks = vbc_tasks::mapped_type;
+
+struct view_building_state {
+    vbc_tasks build_tasks;
+};
+
+query_state& vb_coordinator_query_state() {
+    using namespace std::chrono_literals;
+    const auto t = 1h;
+    static timeout_config tc{ t, t, t, t, t, t, t };
+    static thread_local client_state cs(client_state::internal_tag{}, tc);
+    static thread_local query_state qs(cs, empty_service_permit());
+    return qs;
+}
+
+class view_building_coordinator : public migration_listener::only_view_notifications {
```

Comment:
Why does the coordinator need to subscribe to migration listener? I would expect migration_listener notifications to be eventually removed altogether., they are superseeded by group0 events and direct RPC from the coordinator.